File: mcspring-api/mcspring-jar-loader/src/main/java/org/springframework/boot/loader/mcspring/McSpringLoader.java
package org.springframework.boot.loader.mcspring;

import org.springframework.boot.loader.JarLauncher;
import org.springframework.boot.loader.archive.Archive;

import java.lang.reflect.Method;
import java.net.URL;
import java.net.URLClassLoader;
import java.util.List;

import lombok.SneakyThrows;

// Packaging is so that it all blends in with the Spring loader
public class McSpringLoader extends JarLauncher {
    
    @SneakyThrows
    public void launch(ClassLoader classLoader) {
        List<Archive> activeArchives = getClassPathArchives();
        Method addURL = URLClassLoader.class.getDeclaredMethod("addURL", URL.class);
        addURL.setAccessible(true);
        for (Archive archive : activeArchives) {
            addURL.invoke(classLoader, archive.getUrl());
        }
    }
    
    @Override
    protected String getMainClass() {
        return "";
    }
}

File: mcspring-api/mcspring-jar-loader/src/main/java/org/springframework/boot/loader/mcspring/McSpringLoader.kt
package org.springframework.boot.loader.mcspring

import org.springframework.boot.loader.JarLauncher
import java.net.URL
import java.net.URLClassLoader

// Packaging is so that it all blends in with the Spring loader
class McSpringLoader : JarLauncher() {

    fun launch(classLoader: ClassLoader?) {
        val activeArchives = classPathArchives
        val addURL = URLClassLoader::class.java.getDeclaredMethod("addURL", URL::class.java)
        addURL.isAccessible = true
        activeArchives.forEach { addURL(classLoader, it.url) }
    }

    override fun getMainClass() = ""
}
